crates/runtime/src/dataconnector/kafka.rs

```diff
@@ -45,62 +45,68 @@ pub enum Error {
         "Missing required parameter: 'kafka_bootstrap_servers'. Specify a value. For details, visit: https://spiceai.org/docs/components/data-connectors/kafka#parameters"
     ))]
     MissingKafkaBootstrapServers,
+
+    #[snafu(display(
+        "Failed to parse 'kafka_schema_infer_num_samples' value: {source}. Ensure it is a positive integer and try again."
+    ))]
+    FailedToParseSchemaInferNumSamples { source: std::num::ParseIntError },
 }
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
 #[derive(Debug)]
 pub struct Kafka {
     kafka_config: KafkaConfig,
+    schema_infer_num_samples: Option<usize>,
 }
 
 impl Kafka {
     #[allow(clippy::needless_pass_by_value)]
     pub fn new(params: Parameters) -> Result<Self> {
         let kafka_config = KafkaConfig {
             brokers: params
-                .get("kafka_bootstrap_servers")
+                .get("bootstrap_servers")
```

Comment:
As params belong to the kafka data connector, params definition changed from `ParameterSpec::runtime` to  `ParameterSpec::component` that automatically includes connector name. Thus  there are NO changes from DX/UX perspective